### ISSUE TITLE
build fails in easy_lock.h for 32-bit ARM platform with sched_yield() available

### ISSUE THREAD

**Issue by insi-eb** [curl/curl#9159]
### I did this

I tried building for an embedded 32bit ARM platform that has sched_yield() with an older gcc (7.3).

The build fails with the following error:
```
In file included from easy.c:89:0:
easy_lock.h: In function 'curl_simple_lock_lock':
easy_lock.h:56:7: error: implicit declaration of function 'sched_yield' [-Werror=implicit-function-declaration]
       sched_yield();
       ^~~~~~~~~~~
easy_lock.h:56:7: warning: nested extern declaration of 'sched_yield' [-Wnested-externs]
```
### I expected the following

Build should not fail.

This patch fixes it by including sched.h
```
diff -ru a/lib/easy_lock.h b/lib/easy_lock.h
--- a/lib/easy_lock.h
+++ b/lib/easy_lock.h
@@ -36,6 +36,9 @@

 #elif defined (HAVE_ATOMIC)
 #include <stdatomic.h>
+#if defined(HAVE_SCHED_YIELD)
+#include <sched.h>
+#endif

 #define curl_simple_lock atomic_bool
 #define CURL_SIMPLE_LOCK_INIT false

```
### curl/libcurl version

7.84.0

### operating system

32-bit Yocto Rocko

uname says
`Linux rpi 4.9.80 #1 SMP Mon May 3 20:57:32 UTC 2021 armv7l armv7l armv7l GNU/Linux`

**Comment by dfandrich** [comment]
This patch was already included in e2e7f54b.

**Comment by bagder** [comment]
Already fixed
